From b4014e3d1d9e38b25f2840e65e2acd757f3e5d41 Mon Sep 17 00:00:00 2001
From: Yi Fan Yu <yifan.yu@windriver.com>
Date: Thu, 1 Apr 2021 13:08:37 -0700
Subject: [PATCH] Skip failing tests due to load variability on YP AB

Skip these tests until AB-INT is solved.

[YOCTO #14296]

Upstream-Status: Inappropriate [OE-Specific]

Signed-off-by: Yi Fan Yu <yifan.yu@windriver.com>

Skip two additional tests due to suspected load variability failures.

[YOCTO #15131]
[YOCTO #15177]

Signed-off-by: Trevor Gamblin <tgamblin@baylibre.com>
---
 Lib/test/_test_multiprocessing.py | 3 +++
 Lib/test/test_time.py             | 2 ++
 2 files changed, 5 insertions(+)

diff --git a/Lib/test/_test_multiprocessing.py b/Lib/test/_test_multiprocessing.py
index 3b4415b..1f94dec 100644
--- a/Lib/test/_test_multiprocessing.py
+++ b/Lib/test/_test_multiprocessing.py
@@ -688,6 +688,7 @@ class _TestProcess(BaseTestCase):
         close_queue(q)
 
     @support.requires_resource('walltime')
+    @unittest.skip('timing related test, dependent on load')
     def test_many_processes(self):
         if self.TYPE == 'threads':
             self.skipTest('test not appropriate for {}'.format(self.TYPE))
@@ -2211,6 +2212,7 @@ class _TestBarrier(BaseTestCase):
         except threading.BrokenBarrierError:
             results.append(True)
 
+    @unittest.skip('timing related test, dependent on load')
     def test_timeout(self):
         """
         Test wait(timeout)
@@ -5208,6 +5210,7 @@ class TestWait(unittest.TestCase):
         time.sleep(period)
 
     @support.requires_resource('walltime')
+    @unittest.skip('timing related test, dependent on load')
     def test_wait_integer(self):
         from multiprocessing.connection import wait
 
diff --git a/Lib/test/test_time.py b/Lib/test/test_time.py
index 9463add..4e0f39d 100644
--- a/Lib/test/test_time.py
+++ b/Lib/test/test_time.py
@@ -536,6 +536,7 @@ class TimeTestCase(unittest.TestCase):
     @unittest.skipIf(
         support.is_wasi, "process_time not available on WASI"
     )
+    @unittest.skip('timing related test, dependent on load')
     def test_process_time(self):
         # process_time() should not include time spend during a sleep
         start = time.process_time()
@@ -549,6 +550,7 @@ class TimeTestCase(unittest.TestCase):
         self.assertTrue(info.monotonic)
         self.assertFalse(info.adjustable)
 
+    @unittest.skip('timing related test, dependent on load')
     def test_thread_time(self):
         if not hasattr(time, 'thread_time'):
             if sys.platform.startswith(('linux', 'win')):
